lib/private/BackgroundJob/JobList.php

```diff
@@ -114,6 +115,35 @@ public function remove($job, $argument = null) {
 			$argument = json_encode($argument);
 			$query->andWhere($query->expr()->eq('argument', $query->createNamedParameter($argument)));
 		}
+
+		$query = $this->connection->getQueryBuilder();
+
+		// Add galera safe delete chunking if using mysql
+		// Stops us hitting wsrep_max_ws_rows when large row counts are deleted
+		if ($this->connection->getDatabasePlatform() instanceof MySQLPlatform) {
+			// Then use chunked delete
+			$max = 100000;
+
+			$query->delete('jobs')
+				->where($query->expr()->eq('class', $query->createNamedParameter($class)));
+			if (!is_null($argument)) {
+				$argument = json_encode($argument);
+				$query->andWhere($query->expr()->eq('argument', $query->createNamedParameter($argument)));
+			}
+			$query->setMaxResults($max);
+
+			do {
+				$deleted = $query->execute();
+			} while ($deleted === $max);
+		} else {
+			// Dont use chunked delete - let the DB handle the large row count natively
+			$query->delete('jobs')
+				->where($query->expr()->eq('class', $query->createNamedParameter($class)));
+			if (!is_null($argument)) {
+				$argument = json_encode($argument);
+				$query->andWhere($query->expr()->eq('argument', $query->createNamedParameter($argument)));
+			}
+		}
 		$query->execute();
```

Comment:
```suggestion

		// Add galera safe delete chunking if using mysql
		// Stops us hitting wsrep_max_ws_rows when large row counts are deleted
		if ($this->connection->getDatabasePlatform() instanceof MySQLPlatform) {
			// Then use chunked delete
			$max = 100000;

			$query->setMaxResults($max);
			do {
				$deleted = $query->execute();
			} while ($deleted === $max);
		} else {
			// Dont use chunked delete - let the DB handle the large row count natively
			$query->execute();
		}
```

What went wrong here :D